ground/src/main/java/com/google/android/ground/MainViewModel.kt

```diff
@@ -131,18 +133,18 @@ constructor(
         .subscribe()
     surveySyncSubscription?.let { disposeOnClear(it) }
 
-    return userRepository
-      .saveUser(user)
-      .andThen(
-        if (termsOfServiceRepository.isTermsOfServiceAccepted) {
-          Observable.just(HomeScreenFragmentDirections.showHomeScreen())
-        } else {
-          termsOfServiceRepository.termsOfService
-            .map { SignInFragmentDirections.showTermsOfService().setTermsOfServiceText(it.text) }
-            .cast(NavDirections::class.java)
-            .switchIfEmpty(Maybe.just(HomeScreenFragmentDirections.showHomeScreen()))
-            .toObservable()
-        }
-      )
+    return runBlocking(Dispatchers.IO) {
```

Comment:
Iiuc this will block as soon as the method is called, not when the returned Observable is subscribed to. To reproduce the current behavior, should we instead wrap the coroutine to make its invocation lazy? Please cmiiw!

Reply:
Yes, I think that's correct. Actually, I think this will block at two different times:

1. It will block immediately when this is called and execute `saveUser` and then return an Observable.
2. It will block again when the returned observable is subscribed to.

So you're right, I think for full parity of behavior we need laziness (to make saveUser also lazy).

Reply:
Sg.. assigned to you for follow-up, please feel free to assign back and/or ping when ready.